packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -2119,4 +2119,41 @@ void main() {
     final AssertionError error = exceptions.first as AssertionError;
     expect(error.message, contains('Only one API should be used to manage SnackBars.'));
   });
+
+  testWidgets('Scaffolds can decline SnackBars', (WidgetTester tester) async {
```

Comment:
It would useful to test the registration logic. Rebuilding with registerWithMessenger: true, or with a new Scaffold (change the Scaffold's key).